### PR TITLE
[Feature] Support saving results and show evaluation results in video level

### PR DESCRIPTION
**Pull request by JingweiZhang12**

### CODE REVIEW

File: mmtrack/evaluation/functional/eval_sot_ope.py [hunk 1]
```diff
@@ -61,10 +61,11 @@ def success_error(gt_bboxes_center: np.ndarray, pred_bboxes_center: np.ndarray,
...
 
-def eval_sot_ope(
-        results,
-        annotations: List[List[np.ndarray]],
-        visible_infos: Optional[List[np.ndarray]] = None) -> Dict[str, float]:
+def eval_sot_ope(results,
+                 annotations: List[List[np.ndarray]],
+                 visible_infos: Optional[List[np.ndarray]] = None,
+                 eval_options: Optional[dict] = None,
+                 save_file_cfg: Optional[dict] = None) -> Dict[str, float]:
```
Comment (GT9505) [inline mmtrack/evaluation/functional/eval_sot_ope.py:68]: `eval_options` and `save_file_cfg` are not used

File: mmtrack/evaluation/functional/eval_sot_ope.py [hunk 2]
```diff
@@ -125,9 +126,17 @@ def eval_sot_ope(
             success_error(norm_gt_bboxes_center, norm_pred_bboxes_center,
                           norm_pixel_offset_th, video_length))
 
+    success_results = np.stack(success_results)
+    precision_results = np.stack(precision_results)
+    norm_precision_results = np.stack(norm_precision_results)
```
Comment (GT9505) [inline mmtrack/evaluation/functional/eval_sot_ope.py:131]:
```suggestion
    success_results = np.stack(success_results) *100
    precision_results = np.stack(precision_results)*100
    norm_precision_results = np.stack(norm_precision_results)*100
```

File: mmtrack/evaluation/functional/eval_sot_ope.py [hunk 3]
```diff
@@ -125,9 +126,17 @@ def eval_sot_ope(
...
     success = np.mean(success_results) * 100
     precision = np.mean(precision_results, axis=0)[20] * 100
     norm_precision = np.mean(norm_precision_results, axis=0)[20] * 100
     eval_results = dict(
-        success=success, norm_precision=norm_precision, precision=precision)
+        success=success,
+        norm_precision=norm_precision,
+        precision=precision,
+        ori_success=success_results * 100,
```
[the remark this answers is not in the document]
Comment (JingweiZhang12) [reply mmtrack/evaluation/functional/eval_sot_ope.py:139]: Suggest still use `ori_success`.
The `per_video_success` may be confusing. The `success_results` has two dimensions: video dimension and threshold dimension.

**Comment by JingweiZhang12** [comment]
Guide about saving evaluation results will be added in the plot PR.

File: docs/en/user_guides/6_analysis_tools.md [hunk 4]
```diff
@@ -156,3 +156,18 @@ Optional arguments:
 
 - `SHOW_INTERVAL`: The interval of show (s).
 - `--not-show`: Whether do not show the images on the fly.
+
+## Show SOT evaluation results in video level
+
+The SOT evaluation results are sorted in video level from largest to smallest by the first metric.
```
Comment (GT9505) [inline docs/en/user_guides/6_analysis_tools.md:162]:
```suggestion
The SOT evaluation results are sorted in video level from largest to smallest by the Success metric.
```

File: mmtrack/utils/misc.py [hunk 5]
```diff
@@ -125,3 +125,56 @@ def max_last2d(input: Tensor) -> Tuple[Tensor, Tensor]:
...
+        show_indices (Optional[Tuple[int, List]], optional): The video indices
+            to be shown. Defaults to None, i.e., all videos.
+
+    Returns:
+        List[List]: The formatted video-level evaluation results. For example:
+            [[`video-2`, 48.2, 49.2, 51.9],
+             [`video-1`, 46.2, 48.2, 50.2]]
+    """
+    all_video_names_str = np.array(video_names, dtype=str)
+    eval_show_results = eval_results
```
Comment (GT9505) [inline mmtrack/utils/misc.py:151]: delete this line

**Review by GT9505 — APPROVED** [review]

**Pull request merged**